train_segmentation.py

```diff
@@ -107,13 +104,17 @@ def create_dataloader(samples_folder, batch_size, num_devices, params):
     num_workers = num_devices * 4 if num_devices > 1 else 4
 
     # Shuffle must be set to True.
-    trn_dataloader = DataLoader(trn_dataset, batch_size=batch_size, num_workers=num_workers, shuffle=True,
+    samples_weight = torch.from_numpy(samples_weight)
+    sampler = torch.utils.data.sampler.WeightedRandomSampler(samples_weight.type('torch.DoubleTensor'),
+                                                             len(samples_weight))
+
+    trn_dataloader = DataLoader(trn_dataset, batch_size=batch_size, num_workers=num_workers, sampler=sampler,
```

Comment:
no shuffle?

Reply:
With the sampler, you can't use shuffle. The samples are drawn based on the weights computed in random order.

Reply:
This sampler is really a nice addition. Thanks.